### PR TITLE
enables dtype32 casting option

### PR DESCRIPTION
**Pull request by shreydesai**
Summary: Creates an option to cast logits to dtype32 for the cross-entropy loss module; the decision to cast or not is now available to output layers.

Differential Revision: D16145651

### CODE REVIEW

**Comment by Titousensei** [comment]
Abandoned
